File: src/types/incident.ts
// ============================================================================
// TIPOS DE INCIDENCIA (dinámicos desde la tabla incident_types)
// ============================================================================

/** Tipo de incidencia dinámico (viene de la tabla incident_types) */
export interface IncidentTypeRecord {
  id: string;
  name: string;
  color: string;
  sortOrder: number;
  createdBy?: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateIncidentTypeInput {
  name: string;
  color?: string;
  sortOrder?: number;
}

/** Colores válidos para badges */
export const INCIDENT_TYPE_BADGE_COLORS = [
  'red', 'orange', 'blue', 'green', 'purple', 'gray', 'yellow',
] as const;

export const INCIDENT_TYPE_COLOR_LABELS: Record<string, string> = {
  red: 'Rojo',
  orange: 'Naranja',
  blue: 'Azul',
  green: 'Verde',
  purple: 'Morado',
  gray: 'Gris',
  yellow: 'Amarillo',
};

// --- Incidencia base (listado) ---

export interface Incident {
  id: string;
  rigId: string;
  incidentType: string;
  description: string;
  createdBy: string;
  createdByName?: string;
  createdAt: string;
  updatedAt: string;
}

// --- Personal involucrado ---

export interface IncidentPersonnelInfo {
  id: string;
  personnelId: string;
  name: string;
  ci?: string;
  position: string;
}

// --- Incidencia con personal (detalle) ---

export interface IncidentWithPersonnel extends Incident {
  personnel: IncidentPersonnelInfo[];
}

// --- Input para crear ---

export interface CreateIncidentInput {
  incidentType: string;
  description: string;
  personnelIds: string[];
}

// --- Respuesta paginada ---

export interface PaginatedIncidents {
  data: Incident[];
  total: number;
  page: number;
  pageSize: number;
  totalPages: number;
}

File: src-tauri/src/models/incident.rs
use serde::{Deserialize, Serialize};

// ============================================================================
// INCIDENT TYPES
// ============================================================================

/// Incident with personnel names resolved (for list/detail responses)
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Incident {
    pub id: String,
    pub rig_id: String,
    pub incident_type: String,
    pub description: String,
    pub created_by: String,
    pub created_by_name: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

/// Full incident with involved personnel
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct IncidentWithPersonnel {
    #[serde(flatten)]
    pub incident: Incident,
    pub personnel: Vec<IncidentPersonnelInfo>,
}

/// Personnel info attached to an incident
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct IncidentPersonnelInfo {
    pub id: String,
    pub personnel_id: String,
    pub name: String,
    pub ci: Option<String>,
    pub position: String,
}

/// Input for creating an incident
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateIncidentInput {
    pub incident_type: String,
    pub description: String,
    /// List of rig_personnel IDs involved
    #[serde(default)]
    pub personnel_ids: Vec<String>,
}

/// Paginated response (reuses pattern from logistics)
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PaginatedIncidents {
    pub data: Vec<Incident>,
    pub total: i64,
    pub page: i64,
    pub page_size: i64,
    pub total_pages: i64,
}
